x-pack/plugins/security_solution/server/lib/detection_engine/routes/signals/open_close_signals_route.ts

```diff
@@ -121,17 +121,22 @@ const updateSignalsStatusByIds = async (
   spaceId: string,
   esClient: ElasticsearchClient
 ) =>
-  esClient.bulk({
+  esClient.updateByQuery({
     index: `${DEFAULT_ALERTS_INDEX}-${spaceId}`,
-    refresh: 'wait_for',
-    body: signalsId.flatMap((signalId) => [
-      {
-        update: { _id: signalId, _index: `${DEFAULT_ALERTS_INDEX}-${spaceId}` },
-      },
-      {
-        script: getUpdateSignalStatusScript(status),
+    // conflicts: options.conflicts,
+    // https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html#_refreshing_shards_2
+    // Note: Before we tried to use "refresh: wait_for" but I do not think that was available and instead it defaulted to "refresh: true"
+    // but the tests do not pass with "refresh: false". If at some point a "refresh: wait_for" is implemented, we should use that instead.
+    refresh: true,
```

Comment:
is allowing `refresh:true` something we want to do?
I remember we did some work to avoid explicitly refreshing indices: https://github.com/elastic/security-team/issues/6561

Reply:
In similar work, done in https://github.com/elastic/kibana/pull/162508, I had the same issue, so had add retry calls to wait until item reindexed and only after this finish request.

It worth to note though, updates in lists/items are not available in UI, only through API calls, so we do not expect them to be called frequently, especially in Serverless. Only list delete available in UI, which is also not expected to be frequent operation

Reply:
Thank you for the review @vitaliidm, I took it from here https://github.com/elastic/kibana/blob/4d6f73257487d3ddd86905a7e0030061524ca639/x-pack/plugins/security_solution/server/lib/detection_engine/routes/signals/open_close_signals_route.ts#L130

Reply:
I have tested it locally and it seems `false` works fine, let's see if CI agrees 🤞